redis/db/redisbp/monitored_client.go

```diff
@@ -109,11 +150,35 @@ func MonitorPoolStats(ctx context.Context, client PoolStatser, name string, tags
 		case <-ticker.C:
 			stats := client.PoolStats()
 			hitsGauge.Set(float64(stats.Hits))
+			statsGauge.With(prometheus.Labels{
+				nameLabel:   name,
+				resultLabel: resultHits,
+			}).Set(float64(stats.Hits))
```

Comment:
So, I'm not sure we should hook into the metrics the same way with this one.

This is going to have a couple problems, first the interval/cadence of updates is going to have a missmatch between this and the scape interval.

I would recommend, if we can, to insert our instrumentation at the actual point of change. For example, for a `hit`, we put a counter there and `Inc()`. This more direct instrumentation is less prone to problems.

There are also a bunch of different metrics here that shouldn't be labels, but rather separate metric names.

For example, if you `sum(redisbp_pool_stats)`, its going to include hits+miss+total, which is probably not what the user wants.

Typically in Prometheus we either have a metric for `status="hit"` + `status="miss"`, this way `sum()` works.

Or we have separate `hits_total` and `requests_total` metrics so to get misses you do `requests_total - hits_total`.

Reply:
the upstream redis library does not provide hooks for us to actually emit counters, so I separated the gauges.